src/gui/properties/trackerlistwidget.cpp

```diff
@@ -365,25 +365,27 @@ void TrackerListWidget::loadTrackers()
 
         const BitTorrent::TrackerInfo data = trackerData.value(trackerURL);
 
-        switch (entry.status()) {
+        const auto status = entry.status();
+        switch (status) {
         case BitTorrent::TrackerEntry::Working:
+        case BitTorrent::TrackerEntry::WorkingWarning:
```

Comment:
Why not print "Working with warnings", "Working partially", "Working, errors", "Working, w/ errors" or similar?

Reply:
Why not just show the error/warning msgs in the status column like utorrent and get rid of messages column.

Reply:
> Why not just show the error/warning msgs in the status column like utorrent and get rid of messages column.

I'm aiming here to hotfix logic error that led to lost messages. UX changes can be done in future PRs.

Reply:
I understand. My question was aimed at @FranciscoPombal

Reply:
@an0n666 @kasper93 

My original comment was written before I knew certain things about how libtorrent exposes connection errors vs. application-level protocol error messages to clients.

IIRC the "message" column is currently only used for application-level protocol messages, i.e., the contents of the `failure_reason` key in the bencoded payload that is sent when the connection is actually successful but the tracker wants to report an error. An example of this are messages like `Torrent is unregistered` or `Invalid passkey`. This is different from underlying connection errors, like `Connection timed out`.

Keeping the status column simpler is better for sorting - so I disagree with showing arbitrary connection error messages in this column.

I think there should be 2 "not working states", that should be distinct in the "status column":
- "Not working - Connection error"
- "Not working - Tracker error" - in the current PR this one is `WorkingWarning`, and just prints "Working" - giving no indication that the user should go check the message.

(these are just example names, feel free to come up with better ones. the important thing here is the distinction.)

In both cases, the error message should be printed to the messages column. That way, the "Status" column can be used to quickly sort and separate between any type of connection failures vs. simply tracker application-level errors. The message column becomes useful in both cases to provide more details about the failure.

Reply:
> Keeping the status column simpler is better for sorting
> 

You didn't get my point. You can still sort the status column just like it is now.

If there's no error, the status column will say "Working"
If it's updating, it'll say "Updating"
If it hasn't been contacted yet, "Not contacted"
If there's an error, the status column will show the error message like this:
Error : [message here]
So basically everything will still be sort-able.

Now this message can be the failure_reason sent by the tracker, or the connection error message. This can be tricky now since libtorrent has made things complicated by introducing multi homed announce. You'll have to account for all endpoints and probably display error messages if none of the endpoints work..Then again there'll be disputes so as to which endpoint messages should be stored and given preference over the other endpoint and not overwritten.
The remedy is probably to store messages per endpoint and display them separately in the UI. But this will just clutter the UI in a way which most users may find ugly!

> so I disagree with showing arbitrary connection error messages in this column.
>
Most clients translate these arbitrary error codes to human readable strings.

For instance Deluge shows human readable error messages like Connection timed out, Target machine refused connection etc..These messages give some idea as to where to issue lies.

Reply:
@an0n666 

> You didn't get my point. You can still sort the status column just like it is now.

> Error : [message here]

Right, if you prefix all arbitrary connection error states with `Error: ` you'll still be able to sort the "Status" column reasonably, that's true. So yeah, that would work as well. My point is that `Working` should really mean "absolutely no problem not even a `failure_reason`", and there should be 2 separate status categories for `Not working` - one for connection errors, and one for tracker errors (i.e. `failure_reason`).

> Now this message can be the failure_reason sent by the tracker, or the connection error message. This can be tricky now since libtorrent has made things complicated by introducing multi homed announce. You'll have to account for all endpoints and probably display error messages if none of the endpoints work..Then again there'll be disputes so as to which endpoint messages should be stored and given preference over the other endpoint and not overwritten.
The remedy is probably to store messages per endpoint and display them separately in the UI. But this will just clutter the UI in a way which most users may find ugly!

In the case of a single endpoint, there is no issue of course. if there are multiple endpoints, I think it's more important to show a `failure_reason`, if there is any, instead of any other connection error from any other endpoint (if there are multiple endpoints yielding `failure_reason`, show the latest one, or the first one). If I'm not mistaken, this is the main purpose of this PR. It's still not perfect, since, ideally, one should be able to see all messages/errors from all endpoints, but it's better than the current situation, where which message is shown depends on the order of arrival of each announce response, which is unpredictable.

The reason why displaying a `failure_reason` from endpoint A is more important than a connection error from endpoint B, assuming you can only show one of them, is because ultimately, in such circumstances, the user will care more about the announce that "made it the furthest" 99% of the time. For example, consider the case where the user has a dual-stack network setup, and is announcing to an IPv4-only tracker. The connection errors shown from the IPV6 endpoint are useless, but the potential `failure_reasons` from the IPv4 endpoint are not.

Reply:
@an0n666 @FranciscoPombal

> I think there should be 2 "not working states", that should be distinct in the "status column":

The thing is in practice user doesn't care about this distinction. If there is an error, torrent will not work and this is all user care about. Details and reason is shown in message. 

> The remedy is probably to store messages per endpoint and display them separately in the UI. But this will just clutter the UI in a way which most users may find ugly!

Exactly. I also like verbose output from "everything", but most users will not care about the specific endpoint. But this could be nicely integrated with a tree with tracker->list_of_endpoints.

> If I'm not mistaken, this is the main purpose of this PR. It's still not perfect, since, ideally, one should be able to see all messages/errors from all endpoints, but it's better than the current situation, where which message is shown depends on the order of arrival of each announce response, which is unpredictable.

To see detailed status of every endpoint we would need specific UI for endpoints as mentioned before. I think it will be needed to be done, but it is outside of this PR scope. Like I said before I intend to fix only unpredictable behavior of messages. And all important messages will be shown.

Also if you think about "torrent" status message this PR covers it the best it can to do with single reported message. 

To quickly recap how my changes work, in order:
* If any of endpoints is working (i.e. tracker is working), message will be empty, unless there is warning reported.
* If all endpoints are not working. It will show all messages received from each endpoint. In practice it will be single message, because tracker is not likely to respond differently to different endpoint for the same request, but if it does it will show all messages separated with `|`. 
* If all endpoints are not working and none of the endpoints received response from tracker. System errors for all endpoints will be displayed, deduplicated. This can get verbose depending on the error, but common errors like timeouts/dns errors are easy to understand for user, and most importantly googlable. 

So with this logic, we will have in message field always **most important** information for user. So he can troubleshoot. When tracker responded with error or issue is somewhere else, the one message that is needed will be displayed.

That said one situation where it will still hid the information is when using dual stack and user might want to know if ipv6 is working. In case if ipv4 endpoint is working, other errors will not be displayed. But again this is _specific endpoint debugging_ situation, not a torrent status/message issue (qBt have already logic to consider torrent working if any of endpoints is working, ignoring others). This is one level deeper. In this case torrent is "working" and 99% of users will be happy. And for this 1%, one can always limit qBt to one endpoint and test it directly :)

Reply:
> To quickly recap how my changes work, in order:
> 
> * If any of endpoints is working (i.e. tracker is working), message will be empty, unless there is warning reported.
> * If all endpoints are not working. It will show all messages received from each endpoint. In practice it will be single message, because tracker is not likely to respond differently to different endpoint for the same request, but if it does it will show all messages separated with `|`.
> * If all endpoints are not working and none of the endpoints received response from tracker. System errors for all endpoints will be displayed, deduplicated. This can get verbose depending on the error, but common errors like timeouts/dns errors are easy to understand for user, and most importantly googlable.
> 

Sounds good to me. Hoping this will get reviewed soon and merged.
ping @glassez @sledgehammer999

Reply:
> Sounds good to me. Hoping this will get reviewed soon and merged.
> ping @glassez @sledgehammer999 

Agreed.